runelite-client/src/main/java/net/runelite/client/plugins/antidrag/AntiDragPlugin.java

```diff
@@ -84,32 +111,76 @@ public void keyTyped(KeyEvent e)
 	@Override
 	public void keyPressed(KeyEvent e)
 	{
-		if (e.getKeyCode() == KeyEvent.VK_SHIFT)
+		if (config.onShiftOnly() && e.getKeyCode() == KeyEvent.VK_SHIFT)
 		{
-			final int delay = config.dragDelay();
-			client.setInventoryDragDelay(delay);
-			setBankDragDelay(delay);
+			clientThread.invoke(() ->
+			{
+				if (!inPvp()) {
+					client.setInventoryDragDelay(config.dragDelay());
+					setBankDragDelay(config.dragDelay());
+				}
+			});
 		}
 	}
 
 	@Override
 	public void keyReleased(KeyEvent e)
 	{
-		if (e.getKeyCode() == KeyEvent.VK_SHIFT)
+		if (config.onShiftOnly() && e.getKeyCode() == KeyEvent.VK_SHIFT)
 		{
-			client.setInventoryDragDelay(DEFAULT_DELAY);
-			setBankDragDelay(DEFAULT_DELAY);
+			clientThread.invoke(() ->
+			{
+				client.setInventoryDragDelay(DEFAULT_DELAY);
+				setBankDragDelay(DEFAULT_DELAY);
+			});
 		}
 	}
 
 	@Subscribe
-	public void onFocusChanged(FocusChanged focusChanged)
+	public void onConfigChanged(ConfigChanged event)
 	{
-		if (!focusChanged.isFocused())
+		if (event.getGroup().equals(CONFIG_GROUP))
+		{
+			clientThread.invoke(() ->
+			{
+				if (!config.onShiftOnly() && !inPvp())
+				{
+					client.setInventoryDragDelay(config.dragDelay());
+					setBankDragDelay(config.dragDelay());
+				}
+				else
+				{
+					client.setInventoryDragDelay(DEFAULT_DELAY);
+					setBankDragDelay(DEFAULT_DELAY);
+				}
```

Comment:
These blocks are duplicated numerous times in this plugin now. Move them to helper methods.